OnionUI/Onion
# MainUI patches: Enable button sound for L1/R1 and patch shutdown call

**This is only tested on MM+, someone please test on MM!**

### **Patch 1**
Patched MainUI binaries to play no sound on L1 and R1 keypress.
Instead play sound on L2 and R2, which are now used for scrolling.

_To test:_

- Open game list
- Press L2/R2 
- List should scroll and button sound should play
- Press L1/R1
- no sound should play
- Long press home buttons
- should not hear more than 1 click 


###  **Patch 2**
Patched MainUI binaries to run 
`mainui_off_order` 
instead of 
`customer/app/axp_test poweroff`
by editing the string at 00151aa8

_To test:_

Test case 1
- Go to Settings
- Shutdown
- **The device shuts down**

Test case 2
- Plug in charging cable
- Wait until you see the charging animation
- Unplug the cable
- **The device shuts down**

This is called from two places:
- Settings -> Shutdown
0x0001761c

```
undefined4 settingsShutdown(void)
{
  if (DAT_001808fc != '\0') {
    FUN_0014c5fc(&DAT_00181968,DAT_00181970,DAT_0017e764);
  }
  puts("poweroff2");
  system("sync");
  system("sleep 1");
  system("mainui_off_order"); //previously system("customer/app/axp_test poweroff")
  return 2;
}
```
- Charging manager
0x000389d4
```
...
    if ((local_2c != 2) && (local_2c != 3)) {
      puts("usb unplug, poweroff");
      system("sync");
      system("mainui_off_order"); // previously system("customer/app/axp_test poweroff")
    }
...
```
